### PR TITLE
arcticbio: Update puzzles.py

### PR DESCRIPTION
**Pull request by freckletonj**
1. Modified Magnets combinator to handle undefined block_positions metadata more gracefully.
2. Extended gen_some_blocks to include block_positions metadata (could use more validation)
3. Added extension array with additional puzzle definitions

### CODE REVIEW

**Comment by freckletonj** [comment]
As of right now some of these puzzles don't quite "work", so, i'm gonna go through and triage a bit, for instance:

```
magnets-5 : output seems to be dropping blocks, eg the last row
invert-colors + move-to-pivot: seems weird, can't put my finger on it
translate(3) + right-align: the translation seems irrelevant since they get right-aligned anyway
```

image

**Comment by arcticbio** [comment]
Without significant rewrite, Magnets will only provide meaningful results with two blocks; any more than two, one runs the risk of the shifted blocks layering on top of another blocks, obfuscating the pattern.  Could potentially be rewritten to select two adjacent blocks and continue the translation.  For now I will comment magnets with the constraint and adjust the puzzle.

I also added color 5 back into the base list, since it appears that we are explicitly removing it from use cases where the value is used for pivot.

Overall, I'm struggling with the limited composability of scaling and shifting functions, since they will frequently overwrite each other or run off the working space; two example puzzles added to this branch to explore the outcome and determine how they should be handled.

**Comment by arcticbio** [comment]
Improved composability and edge case handling to allow for more of the puzzles to work with higher degree of consistency without human intervention / oversight.

There are a few new puzzles in place for general review, and I will now go back to the original list, review through the lens of these more robust translation and initialization functions, and bring back in those which are now more transparent and consistent.

File: arc_like/puzzles.py [hunk 1]
```diff
@@ -34,6 +35,15 @@ class Sequence:
 # clean block becomes the input, so we `swap` them after noising.
 Combinator = Callable[[Sequence], Sequence]
 
+def find_all_contiguous_starts(available_positions, size):
```
Comment (freckletonj) [inline arc_like/puzzles.py:37]: :fire: awesome idea

**Comment by freckletonj** [comment]
Thank you @arcticbio , this is looking really awesome!
I'll spend some time tomorrow going through this code too. I really like the `contiguous blocks` idea :heart:

**Comment by arcticbio** [comment]
Hopefully this give you a good starting point for the weekend and maybe a good point to merge the branch.  Puzzle variety is expanded, and known error conditions are resolved.  

I ran all puzzles with 50 samples each over several iterations with floating seed to allow for initial load testing, and there were no errors; spot check review did not indicate any glaring output errors.

**Comment by freckletonj** [comment]
@arcticbio Thank you!  I just fixed up some merge conflicts, but other than that this looks awesome!
I like that you robustified it too, Merging!

**Pull request merged**